docs/standard/serialization/system-text-json-how-to.md

```diff
@@ -0,0 +1,748 @@
+
+```json
+{
+  "Date": "2019-08-01T00:00:00-07:00",
+  "TemperatureC": 25,
+  "Summary": "Hot"
+}
+```
+
+This behavior is intended to help prevent accidental exposure of data in a derived runtime-created type.
+
+To serialize the properties of the derived type, use one of the following approaches:
+
+* Call an overload of `Serialize` that lets you specify the type at runtime:
+
+  ```csharp
+  json = JsonSerializer.Serialize(weatherForecast, weatherForecast.GetType());
+  ```
+
+* Declare the object to be serialized as `object`.
+
+  ```csharp
+  json = JsonSerializer.Serialize<object>(weatherForecast);
+  ```
+
+In the preceding example scenario, both approaches cause the `WindSpeed` property to be included in the JSON output:
+
+```json
+{
+  "Date": "2019-08-01T00:00:00-07:00",
+  "TemperatureC": 25,
+  "Summary": "Hot",
+  "WindSpeed": 35
+}
+```
+
+## Allow overflow JSON to round-trip
+
+While deserializing, you might receive data in the JSON that is not represented by properties of the target type. For example suppose your target type is this:
+
+```csharp
+class WeatherForecast
+{
+    public DateTimeOffset Date { get; set; }
+    public int TemperatureC { get; set; }
+    public string Summary { get; set; }
+}
+```
+
+And the JSON to be deserialized is this:
+
+```json
+{
+  "Date": "2019-08-01T00:00:00-07:00",
+  "temperatureC": 25,
+  "Summary": "Hot",
+  "DatesAvailable": [
+    "2019-08-01T00:00:00-07:00",
+    "2019-08-02T00:00:00-07:00"
+  ],
+  "SummaryWords": [
+    "Cool",
+    "Windy",
+    "Humid"
+  ]
+}
+```
+
+If you deserialize the JSON shown into the type shown, the `DatesAvailable` and `SummaryWords` properties have nowhere to go and are lost. To capture extra data such as these properties, use the [[JsonExtensionData]](xref:System.Text.Json.Serialization.JsonExtensionDataAttribute) attribute on a `Dictionary<string,object>` or `Dictionary<string,JsonElement>` property:
+
+```csharp
+class WeatherForecast
+{
+    public DateTimeOffset Date { get; set; }
+    public int TemperatureC { get; set; }
+    public string Summary { get; set; }
+    [JsonExtensionData]
+    public Dictionary<string, object> ExtensionData { get; set; }
+}
+```
+
+When you deserialize the JSON shown earlier into this sample type, the extra data becomes key-value pairs of the `ExtensionData` property:
+
+|Property |Value  |Notes  |
+|---------|---------|---------|
+| Date    | 8/1/2019 12:00:00 AM -07:00||
+| TemperatureC| 0 | Case-sensitive mismatch (`temperatureC` in the JSON), so the property isn't set. |
+| Summary | Hot ||
+| ExtensionData | temperatureC: 25 |Since the case didn't match, this JSON property is an extra and becomes a key-value pair in the dictionary.|
+|| DatesAvailable:<br>  8/1/2019 12:00:00 AM -07:00<br>8/2/2019 12:00:00 AM -07:00 |Extra property from the JSON becomes a key-value pair, with an array as the value object.|
```

Comment:
the first column is empty by default in these two rows?

Reply:
Yes, the intent is to represent two additional entries in the Dictionary, rather than two additional properties.  I'm open to suggestions for alternative ways of portraying that. The notes explain that intent.